lib_test/test.ml

```diff
@@ -0,0 +1,242 @@
+  let req : Slack_t.list_usergroups_req =
+    { include_count = None; team_id = Some team_id; include_users = Some true; include_disabled = Some true }
+  in
+  match%lwt Api.list_usergroups ~ctx ~req with
+  | Ok res ->
+    let json =
+      res |> Slack_j.string_of_list_usergroups_res |> Yojson.Basic.from_string |> Yojson.Basic.pretty_to_string
+    in
+    print_endline json;
+    Lwt.return_unit
+  | Error e ->
+    log#error "failed to update users of usergroup %s: %s" team_id (Slack_j.string_of_slack_api_error e);
+    Lwt.return_unit
+
+let list_usergroup_users_usergroup_id_list = [ "S04NV4DF0LQ" ]
+let process_list_usergroup_users usergroup =
+  Printf.printf "list_usergroup_users_of--------usergroup_id: %s--------\n" usergroup;
+  let ctx = Context.empty_ctx () in
+  let usergroup : Slack_t.list_usergroup_users_req = { usergroup; team_id = None; include_disabled = Some true } in
+  match%lwt Api.list_usergroup_users ~ctx ~usergroup with
+  | Ok res ->
+    let json =
+      res |> Slack_j.string_of_list_usergroup_users_res |> Yojson.Basic.from_string |> Yojson.Basic.pretty_to_string
+    in
+    print_endline json;
+    Lwt.return_unit
+  | Error e ->
+    log#error "failed to get users of usergroup %s: %s" usergroup.usergroup (Slack_j.string_of_slack_api_error e);
+    Lwt.return_unit
+
+let list_users_cursors_list = [ "dXNlcjpVMDQ2WE4wTTJSNQ==" ]
+let process_list_users cursor =
+  Printf.printf "list_users--------cursor: %s--------\n" cursor;
+  let ctx = Context.empty_ctx () in
+  let req : Slack_t.list_users_req = { cursor = Some cursor; include_locale = None; limit = None; team_id = None } in
+  match%lwt Api.list_users ~ctx ~req with
+  | Ok res ->
+    let json = res |> Slack_j.string_of_list_users_res |> Yojson.Basic.from_string |> Yojson.Basic.pretty_to_string in
+    print_endline json;
+    Lwt.return_unit
+  | Error e ->
+    log#error "failed to get users at cursor %s: %s" cursor (Slack_j.string_of_slack_api_error e);
+    Lwt.return_unit
+
+let mock_slack_event_dir = "mock-slack-events"
+
+let get_mock_slack_events () =
+  List.map (Filename.concat mock_slack_event_dir) (get_sorted_files_from mock_slack_event_dir)
+
+let process_events path =
+  Printf.printf "===== file %s =====\n" path;
+  try
+    let notification = Slack_j.event_callback_notification_of_string (get_local_file path) in
+    let json =
+      notification
+      |> Slack_j.string_of_event_callback_notification
+      |> Yojson.Basic.from_string
+      |> Yojson.Basic.pretty_to_string
+    in
+    printf "%s\n" json;
+    Lwt.return_unit
+  with
+  | Yojson.Json_error e ->
+    printf "failed to parse slack event json %s due to: %s\n" path e;
+    Lwt.return_unit
+  | e ->
+    printf "failed to process slack event %s due to:\n%s\n" path (Printexc.to_string e);
+    Lwt.return_unit
+
+let () =
+  let slack_events = get_mock_slack_events () in
+  Lwt_main.run
+    (let%lwt () = Lwt_list.iter_s process_send_msg simple_text_msg_cases in
```

Comment:
I didn't look too closely but I think the tests can use `iter_p` to run quicker concurrently

```suggestion
    (let%lwt () = Lwt_list.iter_p process_send_msg simple_text_msg_cases in
```

Reply:
Since we're doing a diff I'm not sure if that's a good idea as we want to always have the same order.  The tests are run only in development and I timed it with the `_s` and `_p`: 
```
$ make clean && make && time make test
# with `_p`: 0m0.655s
# with `_s`: 0m0.658s
```
I'm not sure if it's worth that small improvement in development

Reply:
that little huh... yes definitely not worth it